### PR TITLE
feat: add details component

### LINKED ISSUE #196: feat(design): add details component
## Description
An abstraction component which renders expanded list view is need to maintain consistancy among the various component.
Here are certain nice to haves:
- A footer with some fields for meta data like task start time, end time etc.
- A a URL field that renders a copy to clip button.
- A tab group for separation of information. We can have vital info in the first on and rest on the other so on and so forth. 
- A fixed sized text box to show sys logs which can be very long (we can put this in a seperate tab group).

## Possible solution(s)
Like in `Form` component, we can use `.fields` to create such component where each field will specify what kind of render will this be like, `url`, or `large text`, `tag`, `footer` on and so forth. `Package-author` can fill in the `.fields` with the data from API and that could be then used to render the values for their respective fields.

### PR DESCRIPTION
**Pull request by JaeAeich**
## Description

Add details component

Fixes #196 

## Checklist

- [ ] My code follows the [contributing guidelines][contributing] of this project.
- [ ] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

### CODE REVIEW

**Review by SalihuDickson — COMMENTED** [review]
LGTM. I just have a few minor fixes, I went ahead and made a PR to avoid the back and forth, review and merge at #204 so i can approve this and unblock you.

**Comment by SalihuDickson** [comment]
hey, @JaeAeich. If this PR isn't done maybe convert it to a draft first to separate it from PRs that need to be reviewed ASAP.

**Comment by JaeAeich** [comment]
@git-anurag-hub @SalihuDickson please review so I can merge. Hope its fine that we can incrementally add better css and responsive support in other PR, maybe someone can take that up as a new issue.

File: apps/documentation/docs/design/components/details.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+### Buttons
+
+The `buttons` property allows you to define an array of buttons along with their configuration. Buttons can be used to trigger specific actions or events within the component.
+
+| Property  | Required | Default     | Type      | Description                                                                                      |
+| --------- | -------- | ----------- | --------- | ------------------------------------------------------------------------------------------------ |
+| `key`     | `true`   |             | `String`  | A unique identifier for the button, used to distinguish between buttons in the configuration.    |
+| `name`    | `true`   |             | `String`  | The display name or label for the button.                                                        |
+| `size`    | `false`  | `'medium'`  | `String`  | The size of the button. Possible values: `'small'`, `'medium'`, `'large'`.                       |
```
Comment (SalihuDickson) [inline apps/documentation/docs/design/components/details.md:71]: the type here shouldn't be string it should be `small | meduim | large`

**Review by SalihuDickson — CHANGES_REQUESTED** [review]

File: apps/documentation/docs/design/components/details.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+| `tabGroup`              | `true`   |         | `String` | Tab group in which all the its children will be rendered |
+| [`Children`](#children) | `true`   |         | `Array`  | Array of the details to be rendered                      |
+
+### children\*
+
+| Property                  | Required | Default | Type                  | Description                                     |
+| ------------------------- | -------- | ------- | --------------------- | ----------------------------------------------- |
+| `key`                     | `true`   |         | `String`              | The unique key of field                         |
+| `label`                   | `true`   |         | `String`              | The label of the information to be rendered.    |
+| `value`                   | `true`   |         | `String` or `Number`  | The info to be rendered                         |
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply apps/documentation/docs/design/components/details.md]: done

File: packages/ecc-utils-design/src/components/details/details.ts [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+interface FooterButton {
+  key: string;
+  name: string;
+  size: "small" | "medium" | "large";
+  variant: "primary" | "success" | "neutral" | "warning" | "danger";
+  outline: boolean;
+  pill: boolean;
+  icon?: {
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply packages/ecc-utils-design/src/components/details/details.ts:41]: yeah I initially tried that but it was giving error as that slot had to be fixed in slot of shoelace. Maybe we can fixed that later if its not acceptable.

File: packages/ecc-utils-design/src/components/details/details.ts [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+  }
+
+  private _renderLongText(child: Children): TemplateResult {
+    const content = html`<div id="${child.key}" class="value large-text">
+      ${child.value}
+    </div>`;
+    return this._renderContainer(child, content, true);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply packages/ecc-utils-design/src/components/details/details.ts]: done

**Review by SalihuDickson — APPROVED** [review]
I'll go ahead and open an issue for the icons, just so we remember to fix it

**Pull request merged**